api/src/main/java/org/apache/gravitino/rel/types/Types.java

```diff
@@ -330,14 +331,31 @@ public String simpleString() {
 
   /** The time type in Gravitino. */
   public static class TimeType extends Type.DateTimeType {
-    private static final TimeType INSTANCE = new TimeType();
+    private static final TimeType INSTANCE = new TimeType(Optional.empty());
```

Comment:
since the precision must be positive, I suggest to use a static parameter to represent the default precision, WDYT?

```java
/** The base type of all date/time types. */
  abstract class DateTimeType extends PrimitiveType {
    /**
     * Indicates that precision for the date/time type was not explicitly set by the user. The value
     * should be converted to the catalog's default precision.
     */
    public static final int DATE_TIME_PRECISION_NOT_SET = -1;
  }
```